File: xwallpaper/src/main/java/vn/loitp/app/activity/customviews/imageview/zoomimageview/ZoomImageViewActivity.java
package vn.loitp.app.activity.customviews.imageview.zoomimageview;

import android.app.Activity;
import android.os.Bundle;

import vn.loitp.app.activity.customviews.layout.zoomlayout.lib.ZoomImageView;
import vn.loitp.app.base.BaseActivity;
import com.loitp.xwallpaper.R;

public class ZoomImageViewActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ZoomImageView zoomImageView = (ZoomImageView) findViewById(R.id.zoom_iv);
        zoomImageView.setImageDrawable(new ColorGridDrawable());

        /*zoomImageView.getEngine().panTo(x, y, true);
        zoomImageView.getEngine().panBy(deltaX, deltaY, true);
        zoomImageView.getEngine().zoomTo(zoom, true);
        zoomImageView.getEngine().zoomBy(factor, true);
        zoomImageView.getEngine().realZoomTo(realZoom, true);
        zoomImageView.getEngine().moveTo(zoom, x, y, true);*/
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected Activity setActivity() {
        return this;
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_zoomimageview;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/imageview/zoom/ZoomImageViewActivity.kt
package vn.loitp.app.activity.customviews.imageview.zoom

import android.os.Bundle
import androidx.core.view.isVisible
import com.loitp.annotation.IsFullScreen
import com.loitp.annotation.LogTag
import com.loitp.core.base.BaseFontActivity
import com.loitp.core.utilities.LUIUtil
import com.loitp.views.iv.zoom.ColorGridDrawable
import kotlinx.android.synthetic.main.activity_zoom_image_view.*
import vn.loitp.app.R

@LogTag("ZoomImageViewActivity")
@IsFullScreen(false)
class ZoomImageViewActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_zoom_image_view
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setupViews()
    }

    private fun setupViews() {
        lActionBar.apply {
            LUIUtil.setSafeOnClickListenerElastic(
                view = this.ivIconLeft,
                runnable = {
                    onBaseBackPressed()
                }
            )
            this.ivIconRight?.isVisible = false
            this.tvTitle?.text = ZoomImageViewActivity::class.java.simpleName
        }
        zoomImageView.setImageDrawable(ColorGridDrawable())

//        zoomImageView.getEngine().panTo(x, y, true);
//        zoomImageView.getEngine().panBy(deltaX, deltaY, true);
//        zoomImageView.getEngine().zoomTo(zoom, true);
//        zoomImageView.getEngine().zoomBy(factor, true);
//        zoomImageView.getEngine().realZoomTo(realZoom, true);
//        zoomImageView.getEngine().moveTo(zoom, x, y, true);
    }
}
